src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -704,6 +704,26 @@ export const selectAccountIdsByAssetId = createCachedSelector(
     `${state.portfolio.connectedWallet?.id}-${filter?.assetId}` ?? 'assetId',
 )
 
+export const selectAccountIdsByChainId = createDeepEqualOutputSelector(
+  selectPortfolioAccounts,
+  (accounts): Record<ChainId, AccountId[]> =>
+    Object.keys(accounts).reduce<Record<ChainId, AccountId[]>>((acc, accountId) => {
+      const chainId = fromAccountId(accountId).chainId
+      if (!acc[chainId]) acc[chainId] = []
+      acc[chainId].push(accountId)
+      return acc
+    }, {}),
+)
+
+export const selectAccountIdsByChainIdFilter = createCachedSelector(
+  selectPortfolioAccounts,
+  selectChainIdParamFromFilter,
+  findAccountsByChainId,
+)(
+  (state: ReduxState, filter) =>
+    `${state.portfolio.connectedWallet?.id}-${filter?.chainId}` ?? 'chainId',
+)
```

Comment:
what about this and get rid of the util?
```suggestion
export const selectAccountIdsByChainIdFilter = createCachedSelector(
  selectAccountIdsByChainId,
  selectChainIdParamFromFilter,
  (accountIdsByChainId, chainId): AccountId[] => {
    if (!chainId) return []
    return accountIdsByChainId[chainId] ?? []
  },
)(
  (state: ReduxState, filter) =>
    `${state.portfolio.connectedWallet?.id}-${filter?.chainId}` ?? 'chainId',
)
```

Reply:
Absolutely agree - was also thinking we should reuse the newly-introduced selector, though probably best to keep the diff minimal even though that's a very sane change. Also confirmed we have no other usages of it.

<img width="845" alt="image" src="https://github.com/shapeshift/web/assets/17035424/03645baf-362e-4b1b-abc8-b5bed09079aa">

Reply:
yeah thought about that as well. suppose there could be rationale for just using the filter if it makes sense over fetching the whole map and looking up by key. they could both serve a purpose if needed, but no need for the extra logic since we can compose them as filter is the subset.

Reply:
561766a082